gpii/node_modules/gpii-oauth2/gpii-oauth2-authz-server/src/AuthServer.js

```diff
@@ -506,4 +515,21 @@ gpii.oauth2.authServer.contributeRouteHandlers = function (that, oauth2orizeServ
         }
     );
 
+    that.expressApp.post('/add-preferences',
+        function(req, res) {
+            var accessToken = gpii.oauth2.parseBearerAuthorizationHeader(req);
+            if (!accessToken) {
+                res.send(401);
+            } else {
+                var client = that.authorizationService.getCredentialClientByAccessToken(accessToken);
+                if (!client) {
+                    res.send(404);
+                } else if (!client.allowAddPrefs) {
```

Comment:
If I'm reading this correctly, I think we are checking that the client is allowed to add prefs here, rather than checking that the token was granted with the permission of adding prefs.